R/explanation.R

```diff
@@ -444,7 +444,12 @@ explain.combined <- function(x, explainer, approach, prediction_zero, n_samples
                          only_return_dt_mat = TRUE, seed = NULL, ...)
   }
 
+  dt_l_keep <<- dt_l
+
   dt_mat <- unique(rbindlist(dt_l))
+  setkey(dt_mat, row_id)
```

Comment:
@Jens The main issue was that setkey was not called on the entire dt_mat (but only on the sub-dt_mat's), such that the ordering of the rows was incorrect.